assets/blocks/MaturePineapple.block

```diff
@@ -0,0 +1,5 @@
+{
+  "basedOn": "SimpleFarming:Pineapple",
+  "displayName": "Mature Pineapple",
+  "tile": "SimpleFarming:MaturePineapple"
```

Comment:
If the block tile name is the same as the `.block` name you can omit the "tile" entry - the default is looking for a block tile by the same name :-)

However, in this case you don't even *have* a block tile simply named "MaturePineapple.png" :-(

Keep in mind that the pineapple *item* is distinct from any pineapple related *block* - you don't want to have a `.block ` for the loose pineapple meant to be held in the player's hand.